services/src/main/java/com/linkedin/kmf/common/TopicFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+import java.util.Map;
+
+
+/**
+ * Constructs the monitored topic if it does not exist.
+ *
+ * Implementations of this class should have a public, no-arg constructor.
```

Comment:
Should we require this class to have a constructor with properties as well? For example, when we create a security-enabled topic, do we need to provide the topic factory with url of the security service which hosts authentication information?